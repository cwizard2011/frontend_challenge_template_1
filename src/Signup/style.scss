.signup-container {
  display: flex;
  flex-direction: column;
  margin: auto;
  justify-content: center;
  align-items: center;
  background-color: #FAFAFA;
  width: 40%;
  margin-top: 100px;
  padding-bottom: 50px;
  box-shadow: 0.5rem 0.5rem 3rem rgba(0,0,0,0.2);
  padding-top: 20px;
}

.input-field-wrapper {
  display: flex;
  flex-direction: column;
}

.signup-label {
  margin-bottom: 10px;
}

.input-field {
  width: 300px;
  height: 30px;
  border-radius: 5px;
  padding-left: 5px;
  margin-bottom: 15px;
}

.sign-up-text {
  font-size: 25px;
}

.signup-button-class {
  width: 200px;
  height: 40px;
  border-radius: 20px;
  cursor: pointer;
  color: #FFF;
  background-color: #f7436b;
  font-size: 16px;
  border-color: #f7436b;
  box-shadow: 0.5rem 0.5rem 3rem rgba(0,0,0,0.2);
  align-self: center;
  outline: none !important;

  &:disabled {
    background-color: gray;
    border-color: gray;
    cursor: not-allowed;
  }
}

.signup-button-class:hover {
  box-shadow: none;
}

.form-class {
  display: flex;
  flex-direction: column;
  width: 60%;
}

.new-account {
  text-align: center;
}

.error-text {
  color: #f7436b;
  margin-top: 5px;
  font-size: 14px;
}
.kep-login-facebook.metro {
  width: 200px;
  height: 40px;
  border-radius: 20px;
  box-shadow: 0.5rem 0.5rem 3rem rgba(0,0,0,0.2);
  cursor: pointer;
  color: #FFF;
  background-color: #3b5998;
  font-size: 16px;
  border-color: #3b5998;
}

.kep-login-facebook:hover {
  box-shadow: none;
}

.error-input-field {
  border-color: #f7436b !important;
}

@media only screen and (max-width: 767px) {
  .signup-container {
    width: 95%;
  }
}