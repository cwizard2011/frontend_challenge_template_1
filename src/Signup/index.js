import React, { Component } from 'react';
import axios from 'axios';
import toastr from 'toastr';
import { Button, Form, FormGroup, Label, Input, FormText, Alert } from 'reactstrap';
import FacebookLogin from 'react-facebook-login';
import { withToastManager } from 'react-toast-notifications'
import Navbar from '../Navbar';
import { validateSignup } from '../utils/validator';
import '../utils/toastrconfig';
import './style.scss';

export class Signup extends Component {
  state = {
    name: '',
    email: '',
    password: '',
    errors: {

    },
    cart_id: '',
    productIncart: [],
    total_price: '0.00',
    isLoggedIn: false,
    disabledButton: false
  }

  async componentDidMount() {
    const getCartID = await localStorage.getItem('cartId');
    if (!getCartID) {
      const cartID = await axios.get('https://backendapi.turing.com/shoppingcart/generateUniqueId');
      localStorage.setItem('cartId', cartID.data.cart_id);
      this.setState({ cart_id: cartID.data.cart_id })
    } else {
      const response = await axios.get(`https://backendapi.turing.com/shoppingcart/${getCartID}`);
      const totalPrice = await axios.get(`https://backendapi.turing.com/shoppingcart/totalAmount/${getCartID}`);
      this.setState({ cart_id: getCartID, productIncart: response.data, total_price: totalPrice.data.total_amount })
    }
  }

  setUserData = (event) => {
    const { errors } = this.state;
    const newErrors = { ...errors, [event.target.name]: '' };
    this.setState({ [event.target.name]: event.target.value, errors: newErrors })
  }

  signUpUser = async (event) => {
    event.preventDefault();
    this.setState({ disabledButton: true })
    const { name, email, password } = this.state;
    const { history, toastManager } = this.props;
    const { errors, isValid } = validateSignup(name, email, password);

    if (!isValid) {
      return this.setState({ errors, disabledButton: false });
    }

    const data = { name, email, password };
    try {
      const response = await axios.post('https://backendapi.turing.com/customers', data);
      const saveUserData = JSON.stringify(response.data);
      localStorage.setItem('user', saveUserData);
      this.setState({ disabledButton: false })
      history.push('/');
      toastManager.add('Signup successful!', { appearance: 'success', autoDismiss: true });
    } catch (error) {
      if (error.response.data.error) {
        toastManager.add(`${error.response.data.error.message}`, { appearance: 'error', autoDismiss: true } );
        this.setState({ errors: {}, disabledButton: false });
      } else {
        toastManager.add('Unable to signup at this moment', { appearance: 'error', autoDismiss: true });
        this.setState({ errors: {}, disabledButton: false });
      }
    }
  }

  componentClicked = () => console.log('Clicked it')

  responseFacebook = async(response) => {
    const { history, toastManager } = this.props;
    const url = 'https://backendapi.turing.com/customers/facebook';
    const options = {
      url,
      data: {
        access_token: response.accessToken
      },
      method: 'POST',
    }
    try {
      const loginFacebook = await axios(options);
      const saveUserData = JSON.stringify(loginFacebook.data);
      localStorage.setItem('user', saveUserData);
      history.push('/');
      toastManager.add('Signup successful!', { appearance: 'success' })
    } catch (error) {
      if (error.response.data.error) {
        toastManager.add(`${error.response.data.error.message}`, { appearance: 'error', autoDismiss: true } );
      } else {
        toastManager.add('Unable to signup with Facebook at this moment', { appearance: 'error', autoDismiss: true });
      }
    }
  }

  render() {
    let fbContent;
    const { history } = this.props;
    const { errors, productIncart, total_price, isLoggedIn, disabledButton } = this.state;
    if (isLoggedIn) {

    } else {
      fbContent = (
        <FacebookLogin
          appId="352854622106208"
          fields="name,email,picture"
          onClick={() => this.componentClicked()}
          callback={(response) => this.responseFacebook(response)}
          cssClass="kep-login-facebook metro"
          />
      )
    }
    return (
      <div>
        <Navbar
          history={history}
          productIncart={productIncart}
          cartPrice={total_price}
        />
        <div className="signup-container">
          <h5 className="sign-up-text">Sign up</h5>
          <Form id="registerForm" className="form-class">
        <FormGroup>
          <Label for="exampleEmail">Name</Label>
          <Input
            type="text"
            name="name"
            id="name"
            placeholder="Name"
            onChange={this.setUserData}
            className={errors.name ? 'error-input-field' : ''}
          />
          {errors.name && <p className="error-text">{errors.name}</p>}
        </FormGroup>
        <FormGroup>
          <Label for="exampleEmail">Email</Label>
          <Input
            className={errors.email ? 'error-input-field' : ''}
            type="email"
            name="email"
            id="email"
            placeholder="Email"
            onChange={this.setUserData}
            />
          {errors.email && <p className="error-text">{errors.email}</p>}
        </FormGroup>
        <FormGroup>
          <Label for="examplePassword">Password</Label>
          <Input
            className={errors.password ? 'error-input-field' : ''}
            type="password"
            name="password"
            id="password"
            placeholder="Password"
            onChange={this.setUserData}
            />
          {errors.password && <p className="error-text">{errors.password}</p>}
        </FormGroup>
        <p className="new-account">Already have an account?
            <span
              onClick={() => history.push('/login')}
              style={{ color: '#de3624', cursor: 'pointer', marginLeft: 5 }}>
              Login
            </span>
          </p>
          <button
            id="btnFormRegister"
            className="signup-button-class"
            onClick={this.signUpUser}
            disabled={disabledButton}
          >
            {disabledButton ? '....' : 'Register'}
          </button>
          <p className="or">Or</p>
          <div style={{ alignSelf: 'center' }}>
            {fbContent}
          </div>
      </Form>
        </div>
      </div>
    )
  }
}

const SignupWithToast = withToastManager(Signup);

export default SignupWithToast;
