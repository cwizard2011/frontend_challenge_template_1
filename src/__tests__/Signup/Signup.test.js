import React from 'react';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Signup } from '../../Signup';

const script = document.createElement('script');
script.type = 'text/javascript';
document.getElementsByTagName('head')[0].appendChild(script);

describe('<Signup />', () => {
  const props = {
    history: {
      location: {
        pathname: ''
      }
    }
  }
  it('renders without crashing', () => {
    const wrapper = mount(<Signup {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should set the email to be the value entered on input change', () => {
    const value = 'randomemail@gmail.com';
    const wrapper = mount(<Signup {...props} />);
    wrapper.find('#email').at(0).simulate('change', {
      target: {
        name: 'email',
        value
      }
    });
    expect(wrapper.state('email')).toBe(value);
  });

  it('should set the password to be the value entered on input change', () => {
    const value = 'qwertyuiop';
    const wrapper = mount(<Signup {...props} />);
    wrapper.find('#password').at(0).simulate('change', {
      target: {
        name: 'password',
        value
      }
    });
    expect(wrapper.state('password')).toBe(value);
  });

  it('should set the name to be the value entered on input change', () => {
    const value = 'testname';
    const wrapper = mount(<Signup {...props} />);
    wrapper.find('#name').at(0).simulate('change', {
      target: {
        name: 'name',
        value
      }
    });
    expect(wrapper.state('name')).toBe(value);
  });

  it('render errors if we submit without passing values', () => {
    const wrapper = mount(<Signup {...props} />);
    wrapper.find('.form-class').at(0).simulate('submit');
    expect(wrapper.state('errors')).toBeTruthy();
  })
});
