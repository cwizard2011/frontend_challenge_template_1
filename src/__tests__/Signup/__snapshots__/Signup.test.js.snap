// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup /> renders without crashing 1`] = `
<Signup
  history={
    Object {
      "location": Object {
        "pathname": "",
      },
    }
  }
>
  <div>
    <NavbarComponent
      cartPrice="0.00"
      history={
        Object {
          "location": Object {
            "pathname": "",
          },
        }
      }
      productIncart={Array []}
    >
      <div>
        <Navbar
          expand="md"
          light={true}
          style={
            Object {
              "backgroundColor": "#FFF",
            }
          }
          tag="nav"
        >
          <nav
            className="navbar navbar-expand-md navbar-light"
            style={
              Object {
                "backgroundColor": "#FFF",
              }
            }
          >
            <NavbarToggler
              onClick={[Function]}
              tag="button"
              type="button"
            >
              <button
                className="navbar-toggler"
                onClick={[Function]}
                type="button"
              >
                <span
                  className="navbar-toggler-icon"
                />
              </button>
            </NavbarToggler>
            <Collapse
              appear={false}
              enter={true}
              exit={true}
              in={false}
              isOpen={false}
              mountOnEnter={false}
              navbar={true}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              tag="div"
              timeout={350}
              unmountOnExit={false}
            >
              <Transition
                appear={false}
                enter={true}
                exit={true}
                in={false}
                mountOnEnter={false}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={350}
                unmountOnExit={false}
              >
                <div
                  className="collapse navbar-collapse"
                  style={Object {}}
                >
                  <Nav
                    navbar={true}
                    tag="ul"
                    vertical={false}
                  >
                    <ul
                      className="navbar-nav"
                    >
                      <p
                        style={
                          Object {
                            "fontFamily": "Montserrat",
                            "marginLeft": 20,
                            "marginTop": 8,
                          }
                        }
                      >
                        Hi
                      </p>
                      <NavItem
                        style={
                          Object {
                            "display": "flex",
                            "flexDirection": "row",
                          }
                        }
                        tag="li"
                      >
                        <li
                          className="nav-item"
                          style={
                            Object {
                              "display": "flex",
                              "flexDirection": "row",
                            }
                          }
                        >
                          <NavLink
                            onClick={[Function]}
                            style={
                              Object {
                                "color": "#f7436b",
                                "cursor": "pointer",
                                "fontFamily": "Montserrat",
                                "fontWeight": "bold",
                              }
                            }
                            tag="a"
                          >
                            <a
                              className="nav-link"
                              onClick={[Function]}
                              style={
                                Object {
                                  "color": "#f7436b",
                                  "cursor": "pointer",
                                  "fontFamily": "Montserrat",
                                  "fontWeight": "bold",
                                }
                              }
                            >
                              Sign in
                            </a>
                          </NavLink>
                        </li>
                      </NavItem>
                      <NavItem
                        style={
                          Object {
                            "display": "flex",
                            "flexDirection": "row",
                          }
                        }
                        tag="li"
                      >
                        <li
                          className="nav-item"
                          style={
                            Object {
                              "display": "flex",
                              "flexDirection": "row",
                            }
                          }
                        >
                          <p
                            style={
                              Object {
                                "fontFamily": "Montserrat",
                                "marginTop": 8,
                              }
                            }
                          >
                            Or
                          </p>
                          <NavLink
                            onClick={[Function]}
                            style={
                              Object {
                                "color": "#f7436b",
                                "cursor": "pointer",
                                "fontFamily": "Montserrat",
                                "fontWeight": "bold",
                              }
                            }
                            tag="a"
                          >
                            <a
                              className="nav-link"
                              onClick={[Function]}
                              style={
                                Object {
                                  "color": "#f7436b",
                                  "cursor": "pointer",
                                  "fontFamily": "Montserrat",
                                  "fontWeight": "bold",
                                }
                              }
                            >
                              Register
                            </a>
                          </NavLink>
                        </li>
                      </NavItem>
                    </ul>
                  </Nav>
                  <Nav
                    className="ml-auto"
                    navbar={true}
                    tag="ul"
                    vertical={false}
                  >
                    <ul
                      className="ml-auto navbar-nav"
                    >
                      <NavItem
                        style={
                          Object {
                            "marginRight": 20,
                          }
                        }
                        tag="li"
                      >
                        <li
                          className="nav-item"
                          style={
                            Object {
                              "marginRight": 20,
                            }
                          }
                        >
                          <NavLink
                            onClick={[Function]}
                            style={
                              Object {
                                "cursor": "pointer",
                                "display": "flex",
                              }
                            }
                            tag="a"
                          >
                            <a
                              className="nav-link"
                              onClick={[Function]}
                              style={
                                Object {
                                  "cursor": "pointer",
                                  "display": "flex",
                                }
                              }
                            >
                              <i
                                className="fas fa-shopping-bag"
                                style={
                                  Object {
                                    "color": "black",
                                    "fontSize": 20,
                                    "position": "absolute",
                                  }
                                }
                              />
                              <div
                                style={
                                  Object {
                                    "backgroundColor": "#f7436b",
                                    "borderRadius": "50%",
                                    "color": "#FFF",
                                    "height": 20,
                                    "marginLeft": 10,
                                    "marginTop": -10,
                                    "position": "absolute",
                                    "textAlign": "center",
                                    "width": 20,
                                  }
                                }
                              >
                                <p
                                  style={
                                    Object {
                                      "fontFamily": "Montserrat",
                                      "fontSize": 14,
                                      "marginTop": 0,
                                    }
                                  }
                                >
                                  0
                                </p>
                              </div>
                            </a>
                          </NavLink>
                        </li>
                      </NavItem>
                      <NavItem
                        tag="li"
                      >
                        <li
                          className="nav-item"
                        >
                          <NavLink
                            onClick={[Function]}
                            style={
                              Object {
                                "cursor": "pointer",
                                "fontFamily": "Montserrat",
                              }
                            }
                            tag="a"
                          >
                            <a
                              className="nav-link"
                              onClick={[Function]}
                              style={
                                Object {
                                  "cursor": "pointer",
                                  "fontFamily": "Montserrat",
                                }
                              }
                            >
                              Your Bag: 
                            </a>
                          </NavLink>
                        </li>
                      </NavItem>
                      <NavItem
                        tag="li"
                      >
                        <li
                          className="nav-item"
                        >
                          <NavLink
                            onClick={[Function]}
                            style={
                              Object {
                                "cursor": "pointer",
                                "fontFamily": "Montserrat",
                              }
                            }
                            tag="a"
                          >
                            <a
                              className="nav-link"
                              onClick={[Function]}
                              style={
                                Object {
                                  "cursor": "pointer",
                                  "fontFamily": "Montserrat",
                                }
                              }
                            >
                              $ 
                              0.00
                               
                            </a>
                          </NavLink>
                        </li>
                      </NavItem>
                    </ul>
                  </Nav>
                </div>
              </Transition>
            </Collapse>
          </nav>
        </Navbar>
        <Navbar
          expand="md"
          light={true}
          style={
            Object {
              "backgroundColor": "black",
            }
          }
          tag="nav"
        >
          <nav
            className="navbar navbar-expand-md navbar-light"
            style={
              Object {
                "backgroundColor": "black",
              }
            }
          >
            <NavbarToggler
              onClick={[Function]}
              tag="button"
              type="button"
            >
              <button
                className="navbar-toggler"
                onClick={[Function]}
                type="button"
              >
                <span
                  className="navbar-toggler-icon"
                />
              </button>
            </NavbarToggler>
            <Collapse
              appear={false}
              enter={true}
              exit={true}
              in={false}
              isOpen={false}
              mountOnEnter={false}
              navbar={true}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              style={
                Object {
                  "justifyContent": "space-between",
                }
              }
              tag="div"
              timeout={350}
              unmountOnExit={false}
            >
              <Transition
                appear={false}
                enter={true}
                exit={true}
                in={false}
                mountOnEnter={false}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={350}
                unmountOnExit={false}
              >
                <div
                  className="collapse navbar-collapse"
                  style={
                    Object {
                      "justifyContent": "space-between",
                    }
                  }
                >
                  <Nav
                    navbar={true}
                    tag="ul"
                    vertical={false}
                  >
                    <ul
                      className="navbar-nav"
                    >
                      <NavItem
                        style={
                          Object {
                            "display": "flex",
                            "flexDirection": "row",
                          }
                        }
                        tag="li"
                      >
                        <li
                          className="nav-item"
                          style={
                            Object {
                              "display": "flex",
                              "flexDirection": "row",
                            }
                          }
                        >
                          <NavLink
                            href="/"
                            style={
                              Object {
                                "color": "#f7436b",
                                "fontFamily": "Montserrat",
                                "fontSize": 20,
                              }
                            }
                            tag="a"
                          >
                            <a
                              className="nav-link"
                              href="/"
                              onClick={[Function]}
                              style={
                                Object {
                                  "color": "#f7436b",
                                  "fontFamily": "Montserrat",
                                  "fontSize": 20,
                                }
                              }
                            >
                              SHOPMATE
                            </a>
                          </NavLink>
                        </li>
                      </NavItem>
                    </ul>
                  </Nav>
                  <Nav
                    className=""
                    navbar={true}
                    tag="ul"
                    vertical={false}
                  >
                    <ul
                      className="navbar-nav"
                    >
                      <NavItem
                        style={
                          Object {
                            "marginRight": 20,
                          }
                        }
                        tag="li"
                      >
                        <li
                          className="nav-item"
                          style={
                            Object {
                              "marginRight": 20,
                            }
                          }
                        >
                          <NavLink
                            onClick={[Function]}
                            style={
                              Object {
                                "cursor": "pointer",
                                "display": "flex",
                              }
                            }
                            tag="a"
                          >
                            <a
                              className="nav-link"
                              onClick={[Function]}
                              style={
                                Object {
                                  "cursor": "pointer",
                                  "display": "flex",
                                }
                              }
                            >
                              <i
                                className="fas fa-shopping-bag"
                                style={
                                  Object {
                                    "color": "#FFF",
                                    "fontSize": 20,
                                    "position": "absolute",
                                  }
                                }
                              />
                              <div
                                style={
                                  Object {
                                    "backgroundColor": "#f7436b",
                                    "borderRadius": "50%",
                                    "color": "#FFF",
                                    "height": 20,
                                    "marginLeft": 10,
                                    "marginTop": -10,
                                    "position": "absolute",
                                    "textAlign": "center",
                                    "width": 20,
                                  }
                                }
                              >
                                <p
                                  style={
                                    Object {
                                      "fontFamily": "Montserrat",
                                      "fontSize": 14,
                                      "marginTop": 0,
                                    }
                                  }
                                >
                                  0
                                </p>
                              </div>
                            </a>
                          </NavLink>
                        </li>
                      </NavItem>
                    </ul>
                  </Nav>
                </div>
              </Transition>
            </Collapse>
          </nav>
        </Navbar>
      </div>
    </NavbarComponent>
    <div
      className="signup-container"
    >
      <h5
        className="sign-up-text"
      >
        Sign up
      </h5>
      <Form
        className="form-class"
        tag="form"
      >
        <form
          className="form-class"
        >
          <FormGroup
            tag="div"
          >
            <div
              className="form-group"
            >
              <Label
                for="exampleEmail"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <label
                  className=""
                  htmlFor="exampleEmail"
                >
                  Name
                </label>
              </Label>
              <Input
                className=""
                id="name"
                name="name"
                onChange={[Function]}
                placeholder="Name"
                type="text"
              >
                <input
                  className="form-control"
                  id="name"
                  name="name"
                  onChange={[Function]}
                  placeholder="Name"
                  type="text"
                />
              </Input>
            </div>
          </FormGroup>
          <FormGroup
            tag="div"
          >
            <div
              className="form-group"
            >
              <Label
                for="exampleEmail"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <label
                  className=""
                  htmlFor="exampleEmail"
                >
                  Email
                </label>
              </Label>
              <Input
                className=""
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                type="email"
              >
                <input
                  className="form-control"
                  id="email"
                  name="email"
                  onChange={[Function]}
                  placeholder="Email"
                  type="email"
                />
              </Input>
            </div>
          </FormGroup>
          <FormGroup
            tag="div"
          >
            <div
              className="form-group"
            >
              <Label
                for="examplePassword"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <label
                  className=""
                  htmlFor="examplePassword"
                >
                  Password
                </label>
              </Label>
              <Input
                className=""
                id="password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
              >
                <input
                  className="form-control"
                  id="password"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                />
              </Input>
            </div>
          </FormGroup>
          <p
            className="new-account"
          >
            Already have an account?
            <span
              onClick={[Function]}
              style={
                Object {
                  "color": "#de3624",
                  "cursor": "pointer",
                  "marginLeft": 5,
                }
              }
            >
              Login
            </span>
          </p>
          <button
            className="signup-button-class"
            disabled={false}
            onClick={[Function]}
          >
            Register
          </button>
          <p
            className="or"
          >
            Or
          </p>
          <div
            style={
              Object {
                "alignSelf": "center",
              }
            }
          >
            <t
              appId="352854622106208"
              callback={[Function]}
              cssClass="kep-login-facebook metro"
              fields="name,email,picture"
              onClick={[Function]}
              size="metro"
              tag="button"
              textButton="Login with Facebook"
              typeButton="button"
            >
              <t
                appId="352854622106208"
                authType=""
                callback={[Function]}
                cookie={false}
                cssClass="kep-login-facebook metro"
                disableMobileRedirect={false}
                fields="name,email,picture"
                isMobile={false}
                language="en_US"
                onClick={[Function]}
                onFailure={null}
                redirectUri="http://localhost/"
                render={[Function]}
                responseType="code"
                returnScopes={false}
                scope="public_profile,email"
                size="metro"
                state="facebookdirect"
                tag="button"
                textButton="Login with Facebook"
                typeButton="button"
                version="2.3"
                xfbml={false}
              >
                <span
                  style={
                    Object {
                      "opacity": 0.6,
                      "transition": "opacity 0.5s",
                    }
                  }
                >
                  <button
                    className="kep-login-facebook metro metro"
                    onClick={[Function]}
                    type="button"
                  >
                    Login with Facebook
                  </button>
                </span>
              </t>
            </t>
          </div>
        </form>
      </Form>
    </div>
  </div>
</Signup>
`;
